Source/Core/Core/IOS/Network/KD/NetKDRequest.cpp

```diff
@@ -143,6 +143,13 @@ IPCCommandResult NetKDRequest::IOCtl(const IOCtlRequest& request)
     // if ya set the IOS version to a very high value this happens ...
     INFO_LOG(IOS_WC24, "NET_KD_REQ: IOCTL_NWC24_REQUEST_SHUTDOWN - NI");
     break;
+          
+   case IOCTL_NWC24_DOWNLOAD_NOW_EX:
+    // Writing a non-zero return value to this ioctl prevents Nintendo Channel crashing.
+    // For more information, see https://bugs.dolphin-emu.org/issues/11978.
+    INFO_LOG(IOS_WC24, "NET_KD_REQ: IOCTL_NWC24_DOWNLOAD_NOW_EX - NI");
+    WriteReturnValue(NWC24::WC24_ERR_FATAL, request.buffer_out);
```

Comment:
Is this the correct return value? Do you have a hwtest or a disassembly of the relevant code in KD?

Reply:
No. I just forced the return code to an error.
I don't have a hwtest, and I'm not sure how to get one.

Reply:
A hwtest is a small homebrew you write and you run on a real console to test its behaviour. I suspect it would be easier to figure out what the proper error code is or how to signal that nothing was downloaded by disassembling the KD module. Either would be fine; I don't feel comfortable with merging this otherwise.